internal/cli/config_set.go

```diff
@@ -25,17 +26,38 @@ func (c *ConfigSetCommand) Run(args []string) int {
 		return 1
 	}
 
+	var configArgs []string
+
+	// If there are no command arguments, check if the command has
+	// been invoked with a pipe like `cat .env | waypoint config set`.
 	if len(c.args) == 0 {
-		fmt.Fprintf(os.Stderr, "config-set requires at least one key=value entry")
-		return 1
+		info, err := os.Stdin.Stat()
+		if err != nil {
+			_, _ = fmt.Fprintf(os.Stderr, "failed to get console mode for stdin")
+			return 1
+		}
+
+		// If there's no pipe, there are no arguments. Fail.
+		if info.Mode()&os.ModeNamedPipe == 0 {
+			_, _ = fmt.Fprintf(os.Stderr, "config set requires at least one key=value entry")
+			return 1
+		}
+
+		scanner := bufio.NewScanner(os.Stdin)
+		for scanner.Scan() {
+			configArgs = append(configArgs, scanner.Text())
```

Comment:
Nitpick: I would change this a bit to set `configArgs := c.args` earlier up and then just append to that here. That way we don’t need the somewhat awkward `else` fall through.

Reply:
Absolutely, that's a more elegant solution.

Reply:
So I just noticed that I'm now directly appending the config variables to `c.args` without creating a new slice `configArgs` at all - but you actually suggested to initialize `configArgs` with `c.Args` and still append the variables to `configArgs`.

Which way would you prefer?

Reply:
I think its fine as-is since we verify `len(c.args) == 0` to guard entry into that path anyways. Thanks!